spec/inputs/jdbc_spec.rb

```diff
@@ -1267,4 +1267,40 @@
       expect(event.get("ranking").to_f).to eq(95.67)
     end
   end
+
+  context "when debug logging and count query raises an error" do
+    let(:settings) do
+      { "statement" => "SELECT * from types_table" }
+    end
+    let(:logger) { double("logger", :debug? => true) }
+    let(:statement_logger) { LogStash::PluginMixins::CheckedCountLogger.new(logger) }
+    let(:value_tracker) { double("value tracker", :set_value => nil, :write => nil) }
+
+    before do
+      db << "INSERT INTO types_table (num, string, started_at, custom_time, ranking) VALUES (1, 'A test', '1999-12-31', '1999-12-31 23:59:59', 95.67)"
+      plugin.register
+      plugin.set_statement_logger(statement_logger)
+      plugin.set_value_tracker(value_tracker)
+      allow(value_tracker).to receive(:value).and_return("bar")
+      allow(statement_logger).to receive(:execute_count).once.and_raise(StandardError.new("foo"))
+    end
+
+    after do
+      plugin.stop
+    end
+
+    it "should not log a debug line with a count key" do
+      expect(logger).to receive(:info).once.with("Disabling count queries as the executing the count SQL raised an error")
+      expect(logger).to receive(:debug).once.with("Executing JDBC query", :statement => settings["statement"], :parameters => {:sql_last_value=>"bar"})
+      plugin.run(queue)
+      event = queue.pop
+      expect(event.get("num")).to eq(1)
```

Comment:
this test is about not debugging with a count key, maybe we can remove the remaining assertions and create a separate test to check the presence of things that we expect?

Reply:
Sure. I can remove the other expectations, as this context block is a copy of the one at line 1240, those expectations have been verified already.
OTOH, it could be argued that it is important to verify that the normal event creation still occurs when the code from this PR is in the call path and when some degree of stubbing is used.

Reply:
Agreed it may be desirable to test that but I'd put it into a separate `it` block that uses the same test setup (thus it is the same scenario we're testing)